test/userController.test.js

```diff
@@ -42,12 +44,19 @@ beforeAll(async () => {
   // set the url so that our server's mongoose connects to the in-memory mongodb and not our real one
   process.env.MONGO_URL = url;
   mongooseConnection = await connectDB(url);
-  testUser = await createSingleUser({
-    ...testUserCredentials,
-    name: 'Ava Smith',
-    username: 'ava123',
-    role: 'user',
-  });
+  //testUser = await createSingleUser({
+  //  ...testUserCredentials,
+  //  name: 'Ava Smith',
+  //  username: 'ava123',
+  //  role: 'user',
+  //});
+  // Create the test admin user with the specified admin credentials
+  //testAdmin = await createSingleUser({
+  //  ...testAdminCredentials,
+  //  name: 'Admin',
+  //  username: 'admin123',
+  //  role: 'admin', // Set the role to 'admin'
+  //});
```

Comment:
We can delete this :+1:  Since we are creating and assigning the `testUser` variable before each test (`beforeEach`) rather than once at the beginning of the test suite here (`beforeAll`). 

The main reason we do that in this case is because we have some tests like the `updateCurrentUser` test where we modify the testUser. If we only created the `testUser` up here in the `beforeAll` then we'd need to adjust any test that happens after `updateCurrentUser` to expect that the `testUser.name` is now "new\_name" for example. But that's called having "brittle tests" where one test affects the input/output of a whole separate test. So to avoid that; we create a brand new testUser before each test.

For the admin user, we could create it here or in `beforeEach` either one would be fine since we don't have any tests that modify the `testAdmin` user. But to keep the test suite easy to follow and extend, it's probably best to just create all our users in the same place. It also makes it easier to handle our pre-test cleanup where we delete all users in the db so that we start with a fresh slate each time.

Reply:
Thank you, Akos! Very good information for me✍️